gap/PolygonalComplexes/animating.gd

```diff
@@ -0,0 +1,537 @@
+#! Therefore a typical workflow to animate a surfaces looks like this:
+#! <Enum>
+#!   <Item>Construct the surface, e.g. the octahedron above.</Item>
+#!   <Item>Configure your animation using the functions in the Sections
+#!   <Ref Subsect="Section_LabelVisibility"/> and <Ref Subsect="Section_LabelColouring"/>.</Item>
+#!   <Item>Write your animation to a file calling the main function <K>DrawSurfaceToJavaScript</K> (see Section <Ref Sect="Section_LabelCoordinatesAndCoreFunctionality"/>).</Item>
+#! </Enum>
+#! The output of the function <K>DrawSurfaceToJavaScript</K> is a html file. The file can be opened with any browser and then the animation is shown directly.
+
+#! @Section Coordinates and Core Functionality
+#! @SectionLabel LabelCoordinatesAndCoreFunctionality
+
+#! This section deals with the main functions. Before the animation (see <K>DrawSurfaceToJavaScript</K>) can be drawn, it is necessary to specify the 3D-coordinates (which means, the [x,y,z] coordinates) of all vertices of the surface which should be animated (see <K>SetVertexCoordinates3D</K>).
+#!
+#! After seting parameters of the vertices, it is necessary to set parameters of the edges.
+#! We demonstrate the problem of outdated parameters
+#! by an example, where we set coordinates
+#! and redefine these mirrored at Zero.
+#! @BeginExampleSession
+#! gap> verticesPositions := [
+#! > [ 1, -1/Sqrt(3.), -1/Sqrt(6.) ],
+#! > [ -1, -1/Sqrt(3.), -1/Sqrt(6.) ],
+#! > [ 0, 2/Sqrt(3.) , -1/Sqrt(6.) ],
+#! > [ 0, 0, 3/Sqrt(6.) ] ];;
+#! gap> verticesPositionsMirrored := -verticesPositions;;
+#! gap> tet := Tetrahedron();;
+#! gap> printRecord := SetVertexCoordiantes3D(tet, verticesPositions, rec());;
+#! gap> printRecord := CalculateParametersOfEdges(tet, printRecord);;
+#! gap> # coordinates of the center of the edge (1,2):
+#! > printRecord.edges[EdgeBetweenVertices(tet, 1, 2)][1];
+#! [ 0., -0.57735, -0.408248 ]
+#! gap> printRecord := SetVertexCoordiantes3D(tet, verticesPositionsMirrored,
+#! > printRecord);;
+#! gap> # coordinates of the edge (1,2) did not change:
+#! > printRecord.edges[EdgeBetweenVertices(tet, 1, 2)][1];
+#! [ 0., -0.57735, -0.408248 ]
+#! gap> DrawSurfaceToJavaScriptCalculate(tet,
+#! > "doc/Tetrahedron_PositionMirrored.html", printRecord, false);;
+#! @EndExampleSession
+#! @InsertChunk Example_OctahedronPositionMirrored
+#! The function <K>DrawSurfaceToJavaScript</K> always calculates the parameters of the edges depending on the current vertices positions.
```

Comment:
By default, ...

Reply:
function -> method
Try to be accurate here since GAP distinguishes between methods and functions.

Reply:
.. calculates the edge locations depending ..